internal/resourceproviders/cache.go

```diff
@@ -2,22 +2,72 @@ package resourceproviders
 
 import (
 	"context"
-	"log"
+	"fmt"
+	"strings"
+	"sync"
 
-	"github.com/Azure/azure-sdk-for-go/profiles/2017-03-09/resources/mgmt/resources"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonids"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/resources/2022-09-01/providers"
 )
 
 // cachedResourceProviders can be (validly) nil - as such this shouldn't be relied on
 var cachedResourceProviders *[]string
+var registeredResourceProviders *map[string]struct{}
+var unregisteredResourceProviders *map[string]struct{}
```

Comment:
Just a thought, do these need to be pointers? IIRC maps are always passed by reference?

Reply:
this is intentional so these panic if they're accessed without being initialized fwiw, since the cache has to be invoked for it to be usable